planner/core/cache.go

```diff
@@ -172,6 +172,11 @@ func (s FieldSlice) Equal(tps []*types.FieldType) bool {
 			(s[i].EvalType() == types.ETInt && mysql.HasUnsignedFlag(s[i].Flag) != mysql.HasUnsignedFlag(tps[i].Flag)) {
 			return false
 		}
+		// When the type is decimal, we should compare the Flen and Decimal.
```

Comment:
Where do we call the `FieldSlice.Equal`?

Reply:
Only used in plan cache. For example, `Execute.getPhysicalPlan` in `planner/core` package.
Besides the comments for the function in line158 says `Currently this is only used in plan cache to invalidate cache when types of variables are different.`